nvmeof-recoverer: fix `reconstructCRUSHMap` and `runNvmeoFJob()` to work correctly

cheolho-kang opened the pull request:
<!-- Thank you for contributing to Rook! -->

<!-- STEPS TO FOLLOW:
  1. Add a description of the changes (frequently the same as the commit description)
  2. Enter the issue number next to "Resolves #" below (if there is no tracking issue resolved, **remove that section**)
  3. Review our Contributing documentation at https://rook.io/docs/rook/latest/Contributing/development-flow/
  4. Follow the steps in the checklist below, starting with the **Commit Message Formatting**.
-->

<!-- Uncomment this section with the issue number if an issue is being resolved
**Issue resolved by this Pull Request:**
Resolves #
--->

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

cheolho-kang committed:
nvmeof-recoverer: fix `reconstructCRUSHMap` for all OSDs in fabric failure domain

- modified for loop to find all OSD devices specified in the `nvmeofstorage` CR
- fixed OSD device identification logic to check both node name and device path

this commit fixes the reconstruction of the crush map using the fabric failure domain.
previously, the bug caused the reconstruction to work only for one fabric device, and leading to find wrong OSD device when multiple nodes had the same device path.

Signed-off-by: cheolho.kang <cheolho.kang@samsung.com>

myoungwon commented on pkg/operator/ceph/nvmeof_recoverer/clustermanager/handler.go:
```diff
@@ -34,18 +34,18 @@ import time
+    devices = json.loads(result.stdout)
+    devices = {device['DevicePath'] for device in devices['Devices']}
+    return devices
 
```
?

myoungwon replied:
두개로 분리 필요

cheolho-kang replied:
수정하였습니다.
handler.go PR과 controller PR을 분리하는게 나을 것 같다는 말씀이실까요?

myoungwon replied:
indentation과 그 외 내용을 분리하는게 좋을 것 같습니다.

cheolho-kang replied:
아! 넵 수정하였습니다

cheolho-kang committed:
nvmeof_recoverer: fix python script indentation

- fixed mixed indentation in nvmeofToolCode

this commit fixes the indentation issue in the nvme-oF device connect job python script where tabs and spaces were mixed.

Signed-off-by: cheolho.kang <cheolho.kang@samsung.com>

cheolho-kang committed:
nvmeof_recoverer: add namespace argument

- modified `runNvmeoFJob()` to accept namespace as an argument

this commit the functions to accept a namespace argument, enabling them to operate on cephCluster namespaces correctly

Signed-off-by: cheolho.kang <cheolho.kang@samsung.com>
